framework/src/loops/ComputeFullJacobianThread.C

```diff
@@ -60,8 +60,8 @@ ComputeFullJacobianThread::computeJacobian()
       for (const auto & kernel : kernels)
         if ((kernel->variable().number() == ivar) && kernel->isImplicit())
         {
-          kernel->subProblem().prepareShapes(jvar, _tid);
-          kernel->computeOffDiagJacobian(jvariable);
+          kernel->prepareShapes(jvar);
```

Comment:
Oh, I see here. We are not introducing anything new then.

Reply:
Right, this is just introducing more encapsulation. I feel like if we have a non-const `subProblem()` getter in our residual objects, we are doing something wrong